lib/widgets/lightbox.dart

```diff
@@ -0,0 +1,243 @@
+      other.src == other.src;
+  }
+
+  @override
+  int get hashCode => Object.hash('_LightboxHeroTag', messageId, src);
+}
+
+/// Builds a [Hero] from an image in the message list to the lightbox page.
+class LightboxHero extends StatelessWidget {
+  const LightboxHero({
+    super.key,
+    required this.accountId,
+    required this.message,
+    required this.src,
+  });
+
+  /// The account the message list belongs to.
+  final int accountId;
+
+  final Message message;
+  final String src;
+
+  @override
+  Widget build(BuildContext context) {
+    return Hero(
+      tag: _LightboxHeroTag(messageId: message.id, src: src),
+
+      // RealmContentNetworkImage needs an ancestor PerAccountStoreWidget, and
+      // during the hero's "flight", we wouldn't otherwise get one.
+      child: PerAccountStoreWidget(accountId: accountId,
+        child: RealmContentNetworkImage(src, filterQuality: FilterQuality.medium)),
+    );
+  }
+}
+
+class _CopyLinkButton extends StatelessWidget {
+  const _CopyLinkButton({required this.url});
+
+  final String url;
+
+  @override
+  Widget build(BuildContext context) {
+    return IconButton(
+      tooltip: 'Copy link',
+      icon: const Icon(Icons.copy),
+      onPressed: () async {
+        copyWithPopup(context: context, successText: const Text('Link copied'),
+          data: ClipboardData(text: url));
+      });
+  }
+}
+
+class _LightboxPage extends StatefulWidget {
+  const _LightboxPage({
+    required this.routeEntranceAnimation,
+    required this.message,
+    required this.src,
+  });
+
+  final Animation routeEntranceAnimation;
+  final Message message;
+  final String src;
+
+  @override
+  State<_LightboxPage> createState() => _LightboxPageState();
+}
+
+class _LightboxPageState extends State<_LightboxPage> {
+  // TODO: Animate entrance/exit of header and footer
+  bool _headerFooterVisible = false;
+
+  void _handleRouteEntranceAnimationStatusChange(AnimationStatus status) {
+    switch (status) {
+      case AnimationStatus.completed:
+        // Route entrance transition complete
+        setState(() {
+          _headerFooterVisible = true;
+        });
+        break;
+      default:
+        setState(() {
+          _headerFooterVisible = false;
+        });
+        break;
+    }
+  }
+
+  void _handleTap() {
+    setState(() {
+      _headerFooterVisible = !_headerFooterVisible;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    widget.routeEntranceAnimation.addStatusListener(_handleRouteEntranceAnimationStatusChange);
+  }
+
+  @override
+  void dispose() {
+    widget.routeEntranceAnimation.removeStatusListener(_handleRouteEntranceAnimationStatusChange);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final themeData = Theme.of(context);
+
+    final appBarBackgroundColor = Colors.grey.shade900.withOpacity(0.87);
+    const appBarForegroundColor = Colors.white;
+
+    PreferredSizeWidget? appBar;
```

Comment:
similarly, just `Widget?`

Reply:
`Scaffold` asks for a `PreferredSizeWidget?` for its `appBar` param: https://api.flutter.dev/flutter/material/Scaffold/appBar.html

Reply:
Huh, I see! :+1:, then.